runtime/src/bank.rs

```diff
@@ -4138,21 +4132,15 @@ impl Bank {
     /// Process a Transaction. This is used for unit tests and simply calls the vector
     /// Bank::process_transactions method
     pub fn process_transaction(&self, tx: &Transaction) -> Result<()> {
-        let batch = self.prepare_batch(std::iter::once(tx));
+        let batch = self.prepare_batch(std::iter::once(tx))?;
         self.process_transaction_batch(&batch)[0].clone()?;
         tx.signatures
             .get(0)
             .map_or(Ok(()), |sig| self.get_signature_status(sig).unwrap())
     }
 
     #[must_use]
-    pub fn process_transactions(&self, txs: &[Transaction]) -> Vec<Result<()>> {
```

Comment:
nit, maybe this function is removed intentionally, otherwise it could encapsulate the change:
```
{
    let batch = self.prepare_batch(txs.iter()).unwrap();
    self.process_transaction_batch(&batch);
}
```
so same change doesn't need to be repeated at multiple places

Reply:
I intentionally removed it but I think it's worth adding back for tests where a panic on batch creation failures would be ok

Reply:
Added this method back along with a `try_` version for when panic is undesirable